athena-db2-as400/src/main/java/com/amazonaws/athena/connectors/db2as400/Db2As400MetadataHandler.java

```diff
@@ -0,0 +1,494 @@
+     * @return
+     * @throws Exception
+     */
+    private Schema getSchema(Connection jdbcConnection, TableName tableName, Schema partitionSchema)
+            throws Exception
+    {
+        String typeName;
+        String columnName;
+        HashMap<String, String> columnNameMap = new HashMap<>();
+        boolean found = false;
+
+        SchemaBuilder schemaBuilder = SchemaBuilder.newBuilder();
+        try (ResultSet resultSet = getColumns(jdbcConnection.getCatalog(), tableName, jdbcConnection.getMetaData());
+             Connection connection = getJdbcConnectionFactory().getConnection(getCredentialProvider());
+             PreparedStatement stmt = connection.prepareStatement(Db2As400Constants.COLUMN_INFO_QUERY)) {
+            stmt.setString(1, tableName.getSchemaName());
+            stmt.setString(2, tableName.getTableName());
+            try (ResultSet dataTypeResultSet = stmt.executeQuery()) {
+                // fetch data types of columns and prepare map with column name and typeName.
+                while (dataTypeResultSet.next()) {
+                    columnName = dataTypeResultSet.getString("COLUMN_NAME");
+                    typeName = dataTypeResultSet.getString("DATA_TYPE");
+                    columnNameMap.put(columnName, typeName);
+                }
+
+                while (resultSet.next()) {
+                    ArrowType columnType = JdbcArrowTypeConverter.toArrowType(
+                            resultSet.getInt("DATA_TYPE"),
+                            resultSet.getInt("COLUMN_SIZE"),
+                            resultSet.getInt("DECIMAL_DIGITS"),
+                            configOptions
+                    );
+                    columnName = resultSet.getString("COLUMN_NAME");
+                    typeName = columnNameMap.get(columnName);
+
+                    /*
+                    If arrow type is struct then convert to VARCHAR, because struct is
+                    considered as Unhandled type by JdbcRecordHandler's makeExtractor method.
+                     */
+                    if (columnType != null && columnType.getTypeID().name().equalsIgnoreCase("Struct")) {
+                        columnType = Types.MinorType.VARCHAR.getType();
+                    }
+
+                    /*
+                     * Converting REAL, DOUBLE, DECFLOAT data types into FLOAT8 since framework is unable to map it by default
+                     */
+                    if ("real".equalsIgnoreCase(typeName) || "double".equalsIgnoreCase(typeName) || "decfloat".equalsIgnoreCase(typeName)) {
+                        columnType = Types.MinorType.FLOAT8.getType();
+                    }
+
+                    /*
+                     * converting into VARCHAR for non supported data types.
+                     */
+                    if ((columnType == null) || !SupportedTypes.isSupported(columnType)) {
+                        columnType = Types.MinorType.VARCHAR.getType();
+                    }
```

Comment:
This should be:

```
if (...) {
}
else if (...) {
}
else if (...) {
}
```

Because we are not expecting or trying to convert the `columnType` multiple times right?

Additionally, split this out into its own function called:

`getSupportedColumnType(...)`